src/util/expr_initializer.cpp

```diff
@@ -307,5 +323,69 @@ optionalt<exprt> nondet_initializer(
   const source_locationt &source_location,
   const namespacet &ns)
 {
-  return expr_initializert<true>(ns)(type, source_location);
+  return expr_initializert(ns)(type, source_location, exprt(ID_nondet));
+}
+
+/// Create a value for type `type`, with all subtype bytes
+/// initialized to the given value.
+/// \param type: Type of the target expression.
+/// \param source_location: Location to record in all created sub-expressions.
+/// \param ns: Namespace to perform type symbol/tag lookups.
+/// \param init_byte_expr: Value to be used for initialization.
+/// \return An expression if a byte-initialized expression of the input type
+///   can be built.
+optionalt<exprt> expr_initializer(
+  const typet &type,
+  const source_locationt &source_location,
+  const namespacet &ns,
+  const exprt &init_byte_expr)
+{
+  return expr_initializert(ns)(type, source_location, init_byte_expr);
+}
+
+/// Builds an expression of the given output type with each of its bytes
+/// initialized to the given initialization expression.
+/// Integer bitvector types are currently supported.
+/// For unsupported types the initialization expression is casted to the
+/// output type.
+/// \param init_byte_expr The initialization expression
+/// \param output_type The output type
+/// \return The built expression
+exprt duplicate_per_byte(const exprt &init_byte_expr, const typet &output_type)
+{
+  if(output_type.id() == ID_unsignedbv || output_type.id() == ID_signedbv)
+  {
+    const size_t size =
+      to_bitvector_type(output_type).get_width() / config.ansi_c.char_width;
```

Comment:
Why not a more modern?

```suggestion
  if(
    const auto bv_type =
      type_try_dynamic_cast<integer_bitvector_typet>(output_type))
  {
    const size_t size = bv_type->get_width() / config.ansi_c.char_width;
```

Reply:
Changed